walk-in-interview/src/test/java/com/google/account/business/data/BusinessDatabaseTest.java

```diff
@@ -0,0 +1,101 @@
+package com.google.account.business.data;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+import com.google.common.collect.ImmutableList;
+import com.google.utils.FireStoreUtils;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.junit.Assert.assertEquals;
+
+/** Tests for {@link BusinessDatabase} class. */
+public final class BusinessDatabaseTest {
+    // TODO(issue/15): Add failure test case
+
+    private static final String TEST_BUSINESS_COLLECTION = "BusinessAccounts";
+    private static final int BATCH_SIZE = 10;
+
+    static BusinessDatabase businessDatabase;
+    static Firestore firestore;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        businessDatabase = new BusinessDatabase();
+        firestore = FireStoreUtils.getFireStore();
+    }
+
+    @Before
+    public void clearCollection() {
+        try {
+            deleteCollection(firestore.collection(TEST_BUSINESS_COLLECTION), BATCH_SIZE);
+        } catch (ExecutionException | InterruptedException e) {
+            System.err.println("Error deleting collection : " + e.getMessage());
+        }
+    }
+
+    @AfterClass
+    public static void tearDownCollection() {
+        try {
+            deleteCollection(firestore.collection(TEST_BUSINESS_COLLECTION), BATCH_SIZE);
+        } catch (ExecutionException | InterruptedException e) {
+            System.err.println("Error tearing down collection : " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void createBusinessAccount_normalInput_success()
+            throws IOException, ExecutionException, InterruptedException {
+        // Arrange.
+        String uid = "dummyBusinessUid";
+
+        String businessName = "testBusinessAccount";
+        List<String> jobs = ImmutableList.of("jobId1", "jobId2");
+        Business expectedBusiness = Business.newBuilder().setName(businessName).setJobs(jobs).build();
+
+        // Act.
+        Future<WriteResult> future = this.businessDatabase.createBusinessAccount(uid, expectedBusiness);
+
+        // Assert.
+
+        // future.get() blocks on response.
+        future.get();
+
+        DocumentReference documentReference = FireStoreUtils.getFireStore()
+                .collection(TEST_BUSINESS_COLLECTION).document(uid);
+        DocumentSnapshot documentSnapshot = documentReference.get().get();
+
+        Business actualBusiness = documentSnapshot.toObject(Business.class);
+
+        assertEquals(expectedBusiness, actualBusiness);
+    }
```

Comment:
Maybe another test that create multiple account? and see if we have multiple?
There could be a bug that we overwrite existing collection when adding new account?
And another test that create duplicated business?

Reply:
there should not be such bug, since the document id of the account is the uid of the account, while firebase auth will generate only unique uid for each account

and what do u mean by "duplicated business"?

Reply:
Since we now changed from "create" to "update" so we probably don't need to test duplicated create. Aka does the database allow createItem(duplicatedId) multiple times.